lib/bgs/service.rb

```diff
@@ -0,0 +1,197 @@
+# frozen_string_literal: true
+
+require 'bgs/exceptions/service_exception'
+
+module BGS
+  class Service
+    include SentryLogging
+    MAX_ATTEMPTS = 3
+
+    def initialize(user)
+      @user = user
+    end
+
+    def create_participant(proc_id, corp_ptcpnt_id = nil)
+      with_multiple_attempts_enabled do
+        service.vnp_ptcpnt.vnp_ptcpnt_create(
+          {
+            vnp_proc_id: proc_id,
+            ptcpnt_type_nm: 'Person',
+            corp_ptcpnt_id: corp_ptcpnt_id
+          }.merge(bgs_auth)
+        )
+      end
+    end
+
+    def create_person(proc_id, participant_id, payload)
+      with_multiple_attempts_enabled do
+        service.vnp_person.vnp_person_create(
+          create_person_params(proc_id, participant_id, payload)
+        )
+      end
+    end
+
+    def create_address(proc_id, participant_id, payload)
+      with_multiple_attempts_enabled do
+        service.vnp_ptcpnt_addrs.vnp_ptcpnt_addrs_create(
+          create_address_params(proc_id, participant_id, payload)
+        )
+      end
+    end
+
+    def create_phone(proc_id, participant_id, payload)
+      with_multiple_attempts_enabled do
+        service.vnp_ptcpnt_phone.vnp_ptcpnt_phone_create(
+          {
+            vnp_proc_id: proc_id,
+            vnp_ptcpnt_id: participant_id,
+            phone_type_nm: 'Daytime',
+            phone_nbr: payload['phone_number'],
+            efctv_dt: Time.current.iso8601
+          }.merge(bgs_auth)
+        )
+      end
+    end
+
+    def service
+      @service ||= BGS::Services.new(
+        external_uid: @user[:icn],
+        external_key: @user[:external_key]
+      )
+    end
```

Comment:
Would it be possible to abstract this to a separate class that the other services (marriage, form686c, etc) don't inherit from? It would make it more obvious that these are external service calls.

```ruby
bgs_service.create_person
```

It wasn't obvious to me at first glance that all of these external calls were being made.

Reply:
It would also cut down the size of this class.